applications/HDF5Application/custom_io/hdf5_element_data_value_io.cpp

```diff
@@ -11,16 +11,35 @@ namespace HDF5
 {
 namespace
 {
-template <class TVariableType, class TFileDataType>
-void SetDataBuffer(TVariableType const& rVariable,
+template <typename TDataType>
+void SetDataBuffer(Variable<TDataType> const& rVariable,
                    std::vector<ElementType*> const& rElements,
-                   Vector<TFileDataType>& rData);
+                   Vector<TDataType>& rData);
 
-template <class TVariableType, class TFileDataType>
-void SetElementDataValues(TVariableType const& rVariable,
-                          Vector<TFileDataType> const& rData,
+void SetDataBuffer(Variable<Vector<double>> const& rVariable,
```

Comment:
To be consistent you could also omit variable names for these prototypes. It's minor but makes maintenance slightly simpler.